src/nvim/eval.c

```diff
@@ -8694,7 +8704,7 @@ char *do_string_sub(char *str, char *pat, char *sub, typval_T *expr, const char
 
   char *ret = xstrdup(ga.ga_data == NULL ? str : ga.ga_data);
   ga_clear(&ga);
-  if (p_cpo == empty_option) {
+  if (p_cpo == empty_string_option) {
```

Comment:
Shouldn't you use `strcmp` here?

Reply:
Nope, we're comparing the literal addresses here since `empty_string_option` often isn't copied